content/guides/weird_characters.adoc

```diff
@@ -0,0 +1,794 @@
+`@` is used to get the current value of a reference. The above example uses
+`@` to get the current value of an <<xref/../../reference/atom#,atom>>, but `@` can
+be applied to other things such as `future` s, `delay` s, `promises` s etc. to
+force computation and potentially block.
+
+== ^ - Metadata
+
+`^` is the metadata marker. Metadata is a map of values (with shorthand option)
+that can be attached to various forms in Clojure. This provides extra information
+for these forms and can b e used for documentation, compilation warnings,
+typehints, and other features.
+[source,clojure]
+----
+user=> (def ^{ :debug true } five 5) ; meta map with single boolean value
+#'user/five
+----
+
+We can access the metadata by the `meta` function which should be executed
+against the declaration itself (rather than the returned value):
+[source,clojure]
+----
+user=> (def ^{ :debug true } five 5)
+#'user/five
+user=> (meta #'five)
+{:ns #<Namespace user>, :name five, :column 1, :debug true, :line 1, :file "NO_SOURCE_PATH"}
+----
+As we have a single value here, we can use a shorthand notation for declaring
+the metadata `^:name` which is useful for flags, as the value will be set to true.
+[source,clojure]
+----
+user=> (def ^:debug five 5)
+#'user/five
+user=> (meta #'five)
+{:ns #<Namespace user>, :name five, :column 1, :debug true, :line 1, :file "NO_SOURCE_PATH"}
+----
+Another use of `^` is for type hints. These are used to tell the compiler what
+type the value will be and allow it to perform type specific optimiztions
+thus potentially making resultant code faster:
+[source,clojure]
+----
+user=> (def ^Integer five 5)
+#'user/five
+user=> (meta #'five)
+{:ns #<Namespace user>, :name five, :column 1, :line 1, :file "NO_SOURCE_PATH", :tag java.lang.Integer}
+----
+We can see in that example the `:tag` property is set.
+
+You can also stak the shorthand notations:
+[source,clojure]
+----
+user=> (def ^Integer ^:debug ^:private five 5)
+#'user/five
+user=> (meta #'five)
+{:ns #<Namespace user>, :name five, :column 1, :private true, :debug true, :line 1, :file "NO_SOURCE_PATH", :tag java.lang.Integer}
+----
+
+* <<xref/../../reference/metadata#,Clojure Official Documentation>>
+* http://en.wikibooks.org/wiki/Learning_Clojure/Meta_Data[Learning Clojure: Meta Data]
+
+== ' - Quote macro
+
+Can be used against symbols as part of a dispatch macro
+(see <<xref/../weird_characters#dispatch,`#`>>). Also used to quote forms
+and prevent their evalutation as with the quote function.
+[source,clojure]
+----
+user=> (1 3 4) ; fails as it tries to evaluate 1 as a function
+
+ClassCastException java.lang.Long cannot be cast to clojure.lang.IFn  user/eval925 (NO_SOURCE_FILE:1)
+user=> '(1 3 4) ; quote
+(1 3 4)
+user=> (quote (1 2 3)) ; using the longer quote method
+(1 2 3)
+user=>
+----
+
+* <<xref/../../reference/special_forms#quote,Clojure Official Documentation>>
+
+== ; - Comment
+
+`;` is a comment. In fact it's a comment *macro* that takes all input from its
```

Comment:
Is this actually true? I don't think the semicolon should be called a macro. `clojure.core/comment` is a macro.